src/dataset-builder/CohortEditor.ts

```diff
@@ -151,27 +153,37 @@ export const CriteriaView = ({ criteria, deleteCriteria, updateCriteria }: Crite
   );
 
 let criteriaCount = 1;
-const createDefaultListCriteria = (listOption: ProgramDataListOption): ProgramDataListCriteria => {
-  return {
-    kind: 'list',
-    listOption,
-    name: listOption.name,
-    id: Number(_.uniqueId('')),
-    count: 100,
-    values: [listOption.values[0]],
-  };
-};
-
-const createDefaultRangeCriteria = (rangeOption: ProgramDataRangeOption): ProgramDataRangeCriteria => {
-  return {
-    kind: 'range',
-    rangeOption,
-    name: rangeOption.name,
-    id: criteriaCount++,
-    count: 100,
-    low: rangeOption.min,
-    high: rangeOption.max,
-  };
+const createDefaultProgramDataCriteria = async (
+  datasetId: string,
+  option: SnapshotBuilderProgramDataOption
+): Promise<ProgramDataRangeCriteria | ProgramDataListCriteria> => {
+  const generatedOptions = await DatasetBuilder().getProgramDataStatistics(datasetId, option);
```

Comment:
Do we need to build a new `DatasetBuilder` every time we call an ajax API? In the catalog code we didn't do that; we would call the API directly e.g. `Catalog.upsertDataset(...)`.

Also while looking at this, I noticed that `DatasetBuilder.retrieveDataset` is currently unused, as we are using `DataRepo().dataset(...)` to fetch a dataset with this data.

Reply:
For those, we wrapped the Catalog code in `Ajax()`, so it still constructed new values. That is a different approach to doing it than we are doing with DataRepo, which leans more into this centralized `Ajax()` method. I think this approach is maybe a little cleaner. If we want, we could register the `DatasetBuilder` code in Ajax and do it that way instead, but either way it involves rebuilding the object.